tests/test_config.py

```diff
@@ -540,15 +545,16 @@ def test_unknown_option(self) -> None:
         with pytest.warns(CoverageWarning, match=msg):
             _ = coverage.Coverage()
 
-    def test_unknown_option_toml(self) -> None:
+    @pytest.mark.parametrize("filename", ["pyproject.toml", ".coveragerc.toml"])
+    def test_unknown_option_toml(self, filename: str) -> None:
         self.make_file(
-            "pyproject.toml",
+            filename,
             """\
             [tool.coverage.run]
             xyzzy = 17
             """,
         )
-        msg = r"Unrecognized option '\[tool.coverage.run\] xyzzy=' in config file pyproject.toml"
+        msg = f"Unrecognized option '\\[tool.coverage.run\\] xyzzy=' in config file {filename}"
```

Comment:
This would be a bit nicer.
```suggestion
        msg = fr"Unrecognized option '\[tool.coverage.run\] xyzzy=' in config file {filename}"
```